### ISSUE TITLE
Agent on Entra Joined machine - worker service error

### ISSUE THREAD

**Issue by astular** [lithnet/access-manager#241]
Hi! I am testing Lithnet agent on an Entra ID joined machine. I have successfully completed the agent instalation via Intune: I have wrapped the agent and deployed it as a Windows app (Win32).
The agent got installed with parameter AUTHMODE=2 and the SERVER parameter.

After reboot there is an error in the agent.log. I have attached the log file.
[agent.log](https://github.com/user-attachments/files/16945422/agent.log)

Thanks in advance for all your support!

Anze

**Comment by ryannewington** [comment]
The agent is trying to get to `https://<amshostname>/api/versions` and something other than the version response from the server is being returned.

Put this URL in a browser on the impacted host, and see what appears.

It should return a 200 response like
```
{
    "versions": [
        "1.0",
        "1.1",
        "3.0"
    ]
}
```

If it doesn't it should give you some clues as to what is happening. Eg going to the wrong host, incorrectly configured WAF or load balancer in front, etc

**Comment by astular** [comment]
I understand now, where the problem is. We are exposing the Access Manager server through Entra ID Application proxy. If I open the URL in a browser, I get a correct 200 response (because the session is correctly authenticated).
But the computer service runs under a Local System account and the access to the URL does not work.

I have now transitioned to using Microsoft Global Secure Access to provide access to the Access Manager server. One thing I had to do is change the Lithnet Access Manager Agent service startup type to Automatic (Delayed startup), because the Global Secure Access agent is not up and running immediately after logon.

One question: is the Lithnet Access Manager Agent service performing any retries in case the server is not accessible or if it gets an incorrect response?
